semgrep-core/src/extracting/Extract.mli

```diff
@@ -0,0 +1,22 @@
+(* should only be extract rules *)
+val extract_nested_lang :
+  match_hook:
+    (string -> Metavariable.bindings -> Parse_info.t list Lazy.t -> unit) ->
+  timeout:float ->
+  timeout_threshold:int ->
+  Rule.extract_rule list ->
+  Xtarget.t ->
+  string list ->
```

Comment:
This is a list of rule ids we're interested in running. We need this information when generating the additional targets as each target has a list of rule ids to run against it. Normally this would be generated for us by the cli, but in this case because we're creating a new input in core we don't have that.

There's probably a type alias I can swap this to, and when I add docs I'll make it more clear.